esphome/components/max6956/max6956.cpp

```diff
@@ -0,0 +1,183 @@
+#include "max6956.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace max6956 {
+
+static const char *const TAG = "max6956";
+
+/// Masks for MAX6956 Configuration register
+const uint32_t MASK_TRANSITION_DETECTION = 0x80;
+const uint32_t MASK_INDIVIDUAL_CURRENT = 0x40;
+const uint32_t MASK_NORMAL_OPERATION = 0x01;
+
+const uint32_t MASK_1PORT_VALUE = 0x03;
+const uint32_t MASK_PORT_CONFIG = 0x03;
+const uint8_t MASK_CONFIG_CURRENT = 0x40;
+const uint8_t MASK_CURRENT_PIN = 0x0F;
+
+/**************************************
+ *    MAX6956                         *
+ **************************************/
+void MAX6956::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up MAX6956...");
+  uint8_t configuration;
+  if (!this->read_reg_(MAX6956_CONFIGURATION, &configuration)) {
+    this->mark_failed();
+    return;
+  }
+
+  write_brightness_global();
+  write_brightness_mode();
+
+  /** TO DO : read transition detection in yaml
+      TO DO : read indivdual current in yaml **/
+  this->read_reg_(MAX6956_CONFIGURATION, &configuration);
+  ESP_LOGD(TAG, "Initial reg[0x%.2X]=0x%.2X", MAX6956_CONFIGURATION, configuration);
+  configuration = configuration | MASK_NORMAL_OPERATION;
+  this->write_reg_(MAX6956_CONFIGURATION, configuration);
+
+  ESP_LOGCONFIG(TAG, "Enabling normal operation");
+  ESP_LOGD(TAG, "setup reg[0x%.2X]=0x%.2X", MAX6956_CONFIGURATION, configuration);
+}
+
+bool MAX6956::digital_read(uint8_t pin) {
+  uint8_t reg_addr = MAX6956_1PORT_VALUE_START + pin;
+  uint8_t value = 0;
+  this->read_reg_(reg_addr, &value);
+  return (value & MASK_1PORT_VALUE);
+}
+
+void MAX6956::digital_write(uint8_t pin, bool value) {
+  ESP_LOGCONFIG(TAG, "digital_write pin[%u]=%u", pin, value);
+  uint8_t reg_addr = MAX6956_1PORT_VALUE_START + pin;
+  this->write_reg_(reg_addr, value);
+}
+
+void MAX6956::pin_mode(uint8_t pin, gpio::Flags flags) {
+  uint8_t reg_addr = MAX6956_PORT_CONFIG_START + (pin - MAX6956_MIN) / 4;
+  uint8_t config = 0;
+  uint8_t shift = 2 * (pin % 4);
+  uint8_t mode = 0;
```

Comment:
You really should use the enum names when making variables, otherwise its harder to follow the code.
```suggestion
  MAX6956GPIOMode mode = 0;
```